queenbee/schema/arguments.py

```diff
@@ -40,22 +41,69 @@ class Parameter(BaseModel):
 
     path: str = Field(
         None,
-        description='load parameters from a file. File can be a JSON / YAML or a text file.'
+        description='Load parameters values from a JSON file.'
     )
 
+    @root_validator
+    def validate_vars(cls, values):
+        """Validate input values."""
+        name = values.get('name')
+        value = values.get('value')
+        path = values.get('path')
+        if value and path:
+            raise ValueError(
+                f'You should either set value or path for parameter {name}.'
+            )
+
+        value = value if value is not None else path
+        if not value or isinstance(value, (int, float)):
+            return values
+        # check if it is a referenced variable
+        ref_var = qbvar.get_ref_variable(value)
+
+        if ref_var:
+            for rv in ref_var:
+                qbvar.validate_ref_variable_format(rv)
+
+        return values
+
+    @property
+    def current_value(self):
+        """Try to get the current value.
+
+        This method checks the ``value`` property first and if it is None it will return
+        the value for ``path``.
+        """
+        return self.value if self.value is not None else self.path
+
+    @property
+    def ref_vars(self) -> Dict[str, List[str]]:
+        """Get referenced variables if any.
+
+        """
+        value = self.current_value
+        if not value:
+            return {}
+
+        ref_var = qbvar.get_ref_variable(value)
+        if not ref_var:
+            return {}
+
+        return {'value': ref_var} if self.value is not None else {'path': ref_var}
+
 
 class Artifact(BaseModel):
-    """Artifact indicates an artifact to place at a specified path"""
+    """Artifact indicates an artifact to be placed at a specified path."""
 
     name: str = Field(
         ...,
-        description='name of the artifact. must be unique within a task\'s '
+        description='Name of the artifact. Must be unique within a task\'s '
         'inputs / outputs.'
     )
 
     location: str = Field(
-        None,
-        description="Name of the Artifact Location to source this artifact from."
+        None,  # is it possible to create an artifact with no artifact location?
```

Comment:
I'm not sure... My instinct tells me no...

Reply:
Thanks. I will make it a required input then.